conans/model/requires.py

```diff
@@ -212,6 +212,8 @@ def aggregate(self, other):
         self.libs |= other.libs
         self.run = self.run or other.run
         self.visible |= other.visible
+        # The force trait is also defined from an override
+        self.force |= other.force or other.override
```

Comment:
This is the core fix of this PR, the rests are tests, output and removing dead code